doc/source/signals.rst

```diff
@@ -0,0 +1,168 @@
+  # returned by the task and the second element is the signal object.
+  print(result_list[0][1].get_value())
+
+Example: Getting an error signals
+---------------------------------
+
+This is a simple example in which a driver gets an error signal caused
+by the failure of ``task()``.
+
+.. code-block:: python
+
+  @ray.remote
+  def task():
+      raise Exception('exception message')
+
+  object_id = task.remote()
+  try:
+      ray.get(object_id)
+  except Exception as e:
+      pass
+  finally:
+      result_list = signal.receive([object_id], timeout=10)
+      # Expected signal is 'ErrorSignal'.
+      assert type(result_list[0][1]) == signal.ErrorSignal
+      # Print the error.
+      print(result_list[0][1].get_error())
+
+
+Example: Sending signals between multiple actors
+------------------------------------------------
+
+This is a more involved example in which two actors ``a1`` and ``a2`` each
+generate five signals, and another actor ``b`` waits to receive all signals
+generated by ``a1`` and ``a2``, respectively. Note that ``b`` recursively calls
+its own method ``get_signals()`` until it gets all signals it expects.
+
+.. code-block:: python
+
+  @ray.remote
+  class ActorSendSignals(object):
+      def send_signals(self, value, count):
+          for i in range(count):
+              signal.send(UserSignal(value + str(i)))
+
+  @ray.remote
+  class ActorGetAllSignals(object):
+      def __init__(self, num_expected_signals, *source_ids):
+          self.received_signals = []
+          self.num_expected_signals = num_expected_signals
+          self.source_ids = source_ids
+
+      def register_handle(self, handle):
+          self.this_actor = handle
+
+      def get_signals(self):
+          new_signals = signal.receive(self.source_ids, timeout=10)
+          self.received_signals.extend(new_signals)
+          if len(self.received_signals) < self.num_expected_signals:
+              self.this_actor.get_signals.remote()
+
+      def get_count(self):
+          return len(self.received_signals)
+
+  # Create two actors to send signals.
+  a1 = ActorSendSignals.remote()
+  a2 = ActorSendSignals.remote()
+  signal_value = 'simple signal'
+  count = 5
+  # Each actor sends five signals.
+  a1.send_signals.remote(signal_value, count)
+  a2.send_signals.remote(signal_value, count)
+
+  # Create an actor that waits for all five signals sent by each actor.
+  b = ActorGetAllSignals.remote(2 * count, *[a1, a2])
+  # Provide actor to its own handle, so it can recursively call itself
+  # to get all signals from a1, and a2, respectively. This enables the actor
+  # execute other methods if needed.
+  ray.get(b.register_handle.remote(b))
+  b.get_signals.remote()
+  # Print total number of signals. This should be 2*count = 10.
+  print(ray.get(b.get_count.remote()))
+
+Note
+----
+
+A failed actor (e.g., an actor that crashed) generates an error message only
+when another actor or task invokes one of its methods.
```

Comment:
That doesn't seem like the behavior that we want. Also, are you sure that that's true? Consider including an example where an actor crashes (e.g., with `sys.exit(-1)`) and show how to get a signal from that because that's the intended use case.